arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -2473,32 +2473,34 @@ bool ExecutionBlockImpl<Executor>::PrefetchTask::rearmForNextCall(
   TRI_ASSERT(!_result);
   _stack = stack;
   // intentionally do not reset _firstFailure
-  auto old = _state.exchange({Status::Pending, false});
+  auto old =
+      _state.exchange({Status::Pending, false}, std::memory_order_release);
   TRI_ASSERT(old.status == Status::Consumed);
   // if the task was abandoned, we want to reschedule it!
   return old.abandoned;
 }
 
 template<class Executor>
 void ExecutionBlockImpl<Executor>::PrefetchTask::waitFor() const noexcept {
+  std::unique_lock<std::mutex> guard(_lock);
   // (1) - this acquire-load synchronizes with the release-store (3)
-  if (_state.load(std::memory_order_acquire).status == Status::Finished) {
+  if (_state.load().status == Status::Finished) {
```

Comment:
Let's restore the memory order here, so that at least the comment is still up-to-date.
```suggestion
  if (_state.load(std::memory_order_acquire).status == Status::Finished) {
```